_posts/2020-06-05-cruise-control.md

```diff
@@ -0,0 +1,213 @@
+---
+layout: post
+title:  "Cluster balancing with Cruise Control"
+date: 2020-06-04
+author: tom_cooper
+---
+
+Kafka allows you to scale your broker infrastructure to cope with massive load, brokers can be added easily to increase capacity. However, new brokers will only play host to new topics partitions created after the broker was added to the cluster. If certain brokers in your existing cluster are currently under heavy load, then adding a new broker to the cluster will not relieve pressure on the others. To do that you would need to [reassign partiton replicas](https://kafka.apache.org/documentation/#basic_ops_partitionassignment) to the new broker. Similarly, imagine that you have a cluster where, for whatever reason, one broker happens to hold a disproportionate number of partition leaders for partitions under heavy load. To solve this issue you would need to figure out which partition are under load, identify other brokers with lower load which hold replicas for that partition, and then reassign the partition leadership to even out the load across the cluster. This is no easy task. 
+
+Enter [Cruise Control](https://github.com/linkedin/cruise-control), a project from Linkedin, which automates the _balancing_ of load across a Kafka cluster. [Last year]({% post_url 2019-08-08-hacking-for-cruise-control %}), Kyle Liberti described how you could manually get Cruise Control working with Strimzi. With the latest [0.18 release](https://github.com/strimzi/strimzi-kafka-operator/releases/tag/0.18.0), Strimzi now provides native Cruise Control support. 
+
+<!--more-->
+
+## Cruise Control overview
+
+Cruise Control started out as an internal project at Linkedin and was used for managing their in-house Kafka clusters. It was open sourced in 2016 and has seen constant development since then. 
+
+Cruise Control consists of a main server implementation that contains several sub-systems for monitoring, analyzing and proposing changes to the Kafka Cluster it monitors. As well as the main server, Cruise Control provides a Kafka metrics reporter implementation that, once installed into the Kafka brokers, records a wide range of metrics in addition to the default metrics provided by the brokers themselves.
```

Comment:
From what I remember, the metrics the metric reporter records are from the Kafka brokers with the exception the the CPU util metric, which is taken by the metric reporter itself. For that reason I think we should reword the last phrase/sentence to something like this:
```suggestion
Cruise Control consists of a main server implementation that contains several sub-systems for monitoring, analyzing and proposing changes to the Kafka Cluster it monitors. As well as the main server, Cruise Control provides a Kafka metrics reporter implementation that, once installed into the Kafka brokers, filters and records metrics provide by the brokers themselves for further analysis by the mainl server.
```
to avoid the phrase ```records a wide range of metrics in addition```